kong/runloop/plugins_iterator.lua

```diff
@@ -149,6 +149,7 @@ local function get_plugin_config(plugin, name, ws_id)
   cfg.route_id    = plugin.route    and plugin.route.id
   cfg.service_id  = plugin.service  and plugin.service.id
   cfg.consumer_id = plugin.consumer and plugin.consumer.id
+  cfg.id          = plugin.id
```

Comment:
I actually had no idea we populated the plugin config table like this--interesting!

I don't have a strong objection to `id`, but what do you think about using `plugin_id` instead? It would be more consistent with the other entities, and I think it seems _slightly_ less likely to conflict with any plugin that may have an `id` field in its config schema.

Reply:
I was planning to add `plugin_custom_name` here on another [PR](https://github.com/Kong/kong/pull/10077). 

After saw the comment here, I was thinking the Plugin entity adds more and more field during its iteration, populating plugin's meta data into config table flatly is not a correct way, it increases the chances of conflict with user's custom plugin.

Perhaps we can fold the plugin's meta data into `meta` field, or even provide different way to access plugin's meta.

```lua
function MyPlugin:access(conf, plugin) 
  -- access plugin's meta
  local plugin_id = plugin.id
  local plugin_custom_name = plugin.custom_name
end
```

Reply:
Using the meta is a bit hacky. I think I prefer adding the plugin as a second parameter.